lib/new_sets.bzl

```diff
@@ -34,6 +34,8 @@ def _make(elements = None):
       A set containing the passed in values.
     """
     elements = elements if elements else []
+    if type(elements) == "depset":
+        elements = elements.to_list()
```

Comment:
I recommend against it: https://docs.bazel.build/versions/master/skylark/performance.html#never-call-depset-to_list
Flattening or iterating over a `depset` should be explicit.

Reply:
@laurentlb the Depsets page says otherwise:

https://g3doc.corp.google.com/devtools/blaze/rules/g3doc/depsets.md#upcoming-changes

> * Depset contents should be retrieved using to_list(), not by iterating over the depset itself. Direct iteration over depsets is deprecated and will be removed. I.e., don't use list(...), sorted(...), or other functions expecting an iterable, on depsets.

Reply:
The page you're citing is talking about how implicit iteration over depsets is deprecated, and the preferred alternative is to do it explicitly with `to_list()`. In this case, a library function is wrapping the call to `to_list()`, so Laurent is concerned that it's effectively being made implicit once again, at least from the POV of the caller of `sets.make()`.

(You could argue that if you're asking to make a hashset out of a depset, that in itself is a form of explicit flattening. Not sure where I stand on that.)

In the previous code, users can run with `--all_incompatible_changes` to know whether or not they have any call sites that do `sets.make(mydepset)` when they really should've been rewritten to not flatten depsets, or at least marked explicitly with `sets.make(mydepset.to_list())`. With this change, `sets.make(mydepset)` will always be allowed with or without `--all_incompatible_changes`.

On the other hand, if we revert this change, the `sets.make(mydepset)` usage will continue to be an error if and only if the incompatible flag is given. In the interest of preserving semantics in this PR I think that's what we should do.

Reply:
I don't have access to the g3doc page, so please pardon me I'm not following the conversation exactly :D 

It would be nice if this function was compatible with `--all_incompatible_changes`, regardless of the arguments passed to it. If we want to make this code not accept depsets, wouldn't it be better to fail when a depset is passed to it and still be compatible with `--all_incompatible_changes`?

Reply:
The g3doc page is here: https://docs.bazel.build/versions/master/skylark/depsets.html#upcoming-changes

Reply:
Edited/corrected my link above. Sorry, I meant to grab the external one in the first place.

Reply:
Thanks for updating the link! :)

Reply:
new_sets.bzl was introduced because we had folks internally using the broken sets.bzl implementation, "dictionary of Bools" is kind of a busted abstraction to recommend to users who need set functionality, and the module lets us expose important operations like union/intersection more cleanly than just using the raw dictionary.